### ISSUE TITLE
add new feature

### ISSUE THREAD

**Issue by geosky** [GeospatialPython/pyshp#14]
it seems that right now when adding new features save() will write the whole file once again, not append, right? adding new features extremely slow and unstable when data size is hugh.

**Comment by GeospatialPython** [comment]
Yes - the bottleneck here is you have to track all of the features in order to write the headers and index files.  Figuring out how to do that efficiently is tricky.

**Comment by karimbahgat** [comment]
No longer relevant after #98 which drops the Editor class, which also fixes part of the issue by enabling streamed writing for unlimited file sizes (writes all entries immediately to temporary files, and merges them with the final header at the end).
